Mlem/Views/Shared/AppLockView.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  AppLockView.swift
+//  Mlem
+//
+//  Created by Sumeet Gill on 2024-01-16.
+//
+
+import Dependencies
+import LocalAuthentication
+import SwiftUI
+
+struct AppLockView: View {
+    @Dependency(\.accountsTracker) var accountsTracker
+
+    @ObservedObject var biometricUnlock: BiometricUnlock
+    @Environment(\.scenePhase) var phase
+    
+    @AppStorage("appLock") var appLock: AppLock = .disabled
+    
+    @State var isErrorVisible: Bool = false
+    @State var isButtonHidden = true
+    @State var isAuthenticating = false
+    @State var alertMessage: String = ""
+
+    var body: some View {
+        @State var logoOffset: CGFloat = 0
+
+        VStack(spacing: 50) {
+            LogoView()
+                .frame(width: 150, height: 150)
+                .animation(.spring(), value: isAuthenticating)
+                .offset(y: isAuthenticating ? 0 : 300)
+            
+            Group {
+                Button {
+                    authenticate()
+                } label: {
+                    HStack {
+                        Spacer()
+                        Text("Unlock Mlem")
+                            .bold()
+                        Spacer()
+                    }
+                    .frame(width: 200)
+                    .padding(15)
+                    .background(.blue)
+                    .foregroundColor(.white)
+                    .clipShape(RoundedRectangle(cornerRadius: 12))
+                }
+            }
+            .frame(maxHeight: .infinity, alignment: .bottom)
+            .opacity(isButtonHidden ? 0 : 1)
+        }
```

Comment:
Using a `ZStack` instead of a `VStack` (see previous comment) and adding `.ignoresSafeArea` here ensures that the logo remains centered on all devices

```suggestion
                    .clipShape(RoundedRectangle(cornerRadius: 12))
                    .padding(.bottom, 50)
                }
            }
            .frame(maxHeight: .infinity, alignment: .bottom)
            .opacity(isButtonHidden ? 0 : 1)
        }
        .ignoresSafeArea()
```